components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -1363,6 +1328,18 @@ else if (getZCTCoords(plane)[2] < timestamps.size()) {
     segments = null;
   }
 
+  // -- ZeissCZI-specific methods --
+
+  public boolean allowAutostitching() {
+    return ((DefaultMetadataOptions) getMetadataOptions()).getBoolean(
```

Comment:
You're right. We could change `setMetadataOptions` to *update* the object rather than reassign it, but any attempt to use a custom implementation of `MetadataOptions` would not work as expected (not on `FormatHandler` instances, anyway).

So, if we don't decide to go for one of your proposed refactorings of the options API, I will apply the `instanceof` fix.